### DISCUSSION TITLE
Ordinals and Gordian Envelopes?

### DISCUSSION THREAD

**Discussion by ChristopherA** [ordinals/ord#1211] [Ideas]
One possible interesting use of the ordinals is to place an Gordian Envelope (a CBOR object) on-chain. Gordian Envelopes support a variety of features, including escrowed encryption, shared secret recovery shares, elision/redaction, proof-of-inclusion, etc. The smallest envelope could be the top of an elided tree of assertions allowing parties to use proof-of-inclusion to demonstrate their assertions were valid.

IETF spec: https://datatracker.ietf.org/doc/draft-mcnally-envelope/

Overview: https://www.blockchaincommons.com/introduction/Envelope-Intro/

Video playlist: https://www.youtube.com/playlist?list=PLCkrqxOY1FbooYwJ7ZhpJ_QQk8Az1aCnG

-- Christopher Allen

**Comment by casey** [comment]
I actually call the script construction `OP_FALSE OP_IF <data pushes OP_ENDIF` used to hold inscriptions an envelope, so it's probably meant to be. Do Gordian Envelopes have a MIME type? Inscription content type is identified by a MIME type, so that would be a prerequisite to making an inscription with one.

**Comment by ChristopherA** [comment]
Yes: https://blockchaincommons.github.io/WIPs-IETF-draft-envelope/draft-mcnally-envelope.html#name-media-type

> The proposed media type [[RFC6838]] for envelope is application/envelope+cbor.
